File: 121.买卖股票的最佳时机.py
#
# @lc app=leetcode.cn id=121 lang=python3
#
# [121] 买卖股票的最佳时机
#

# @lc code=start
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit = 0
        min_price =  prices[0]
        for price in prices:
            max_profit = max(max_profit, price - min_price)
            min_price = min(price, min_price)
        return max_profit
# @lc code=end

File: leetcode-go/121.买卖股票的最佳时机.go
/*
 * @lc app=leetcode.cn id=121 lang=golang
 *
 * [121] 买卖股票的最佳时机
 */
// package main

// import "fmt"

// func main() {
// 	prices := []int{7, 1, 5, 3, 6, 4}
// 	res := maxProfit(prices)
// 	fmt.Println(res)
// }

// @lc code=start
func maxProfit(prices []int) (ans int) {
	minPrice := prices[0]
	for _, p := range prices {
		ans = max(ans, p-minPrice)
		minPrice = min(minPrice, p)
	}
	return
}

// @lc code=end
